pkg/okctl/okctl.go

```diff
@@ -765,30 +758,72 @@ https://www.passwordstore.org/
 		_ = k.Store(keyring.KeyTypeUserPassword, password)
 	}
 
-	var credentialsRetriever aws.Retriever
+	authStore := aws.NewIniPersister(aws.NewFileSystemIniStorer(
+		config.DefaultClusterAwsConfig,
+		config.DefaultClusterAwsCredentials,
+		path.Join(appDir, config.DefaultCredentialsDirName, o.RepoStateWithEnv.GetClusterName()),
+		o.FileSystem,
+	))
+
+	return aws.New(authStore, aws.NewAuthSAML(
+		o.RepoStateWithEnv.GetCluster().AWSAccountID,
+		o.RepoStateWithEnv.GetMetadata().Region,
+		scrape.New(),
+		aws.DefaultStsProvider,
+		aws.Interactive(o.Username(), storedPassword, fn),
+	)), nil
+}
 
-	switch o.CredentialsType {
-	case context.CredentialsTypeAccessKey:
-		credentialsRetriever = aws.NewAuthEnvironment(o.RepoStateWithEnv.GetMetadata().Region, os.Getenv)
-	default:
-		credentialsRetriever = aws.NewAuthSAML(
-			o.RepoStateWithEnv.GetCluster().AWSAccountID,
-			o.RepoStateWithEnv.GetMetadata().Region,
-			scrape.New(),
-			aws.DefaultStsProvider,
-			aws.Interactive(o.Username(), storedPassword, fn),
-		)
+func (o *Okctl) getGithubAuthenticator() (*github.Auth, error) {
+	if o.GithubCredentialsType == context.GithubCredentialsTypeToken {
+		return github.New(
+			github.NewInMemoryPersister(),
+			&http.Client{Timeout: authenticatorTimeout},
+			github.NewAuthEnvironment(os.Getenv),
+		), nil
 	}
 
-	gh := github.New(
+	defaultRing, err := keyring.DefaultKeyring()
+	if err != nil {
+		return nil, fmt.Errorf(`unable to create a keyring. It is possible no valid backends were found 
+on your system, take a look at this site for valid options:
+https://github.com/99designs/keyring#keyring
+
+On linux pass works well, for instance:
+https://www.passwordstore.org/
+
+%w`, err)
+	}
+
+	k, err := keyring.New(defaultRing, o.Debug)
+	if err != nil {
+		return nil, err
+	}
+
+	return github.New(
 		github.NewKeyringPersister(k),
-		&http.Client{
-			Timeout: 5 * time.Second, // nolint: gomnd
-		},
+		&http.Client{Timeout: authenticatorTimeout},
 		github.NewAuthDeviceFlow(github.DefaultGithubOauthClientID, github.RequiredScopes()),
-	)
+	), nil
+}
+
+// newCredentialsProvider knows how to load credentials
+func (o *Okctl) newCredentialsProvider() error {
+	if o.NoInput {
```

Comment:
I guess you need to add something like: `&& github_device || aws_saml` for this if test to make sense?

Reply:
Removed the o.NoInput flag. We can reintroduce it when its needed